src/test/java/org/elasticsearch/index/query/CommonTermsQueryTest.java

```diff
@@ -0,0 +1,87 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queries.ExtendedCommonTermsQuery;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.settings.Settings.Builder;
+import org.elasticsearch.index.mapper.ContentPath;
+import org.elasticsearch.index.mapper.Mapper;
+import org.elasticsearch.index.mapper.core.DateFieldMapper;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class CommonTermsQueryTest extends BaseQueryTestCase<CommonTermsQueryBuilder> {
+
+    @Override
+    protected CommonTermsQueryBuilder createTestQueryBuilder() {
+        CommonTermsQueryBuilder query = new CommonTermsQueryBuilder("field", "text"); 
+
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomAsciiOfLengthBetween(1, 10));
+        }
+        
+        return query;
+    }
+
+    @Override
+    protected Query createExpectedQuery(CommonTermsQueryBuilder queryBuilder, QueryParseContext context) throws IOException {
+        ExtendedCommonTermsQuery expectedQuery;
+
+        DateFieldMapper.Builder fieldMapperBuilder = new DateFieldMapper.Builder(queryBuilder.fieldName());
+        DateFieldMapper fieldMapper = fieldMapperBuilder.build(new Mapper.BuilderContext(Builder.EMPTY_SETTINGS.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).build(), new ContentPath()));
+
+        BooleanClause.Occur highFreqOccur = queryBuilder.highFreqOperator().equals(CommonTermsQueryBuilder.Operator.AND) ? BooleanClause.Occur.MUST : BooleanClause.Occur.SHOULD;
+        BooleanClause.Occur lowFreqOccur = queryBuilder.lowFreqOperator().equals(CommonTermsQueryBuilder.Operator.AND) ? BooleanClause.Occur.MUST : BooleanClause.Occur.SHOULD;
+
+        expectedQuery = new ExtendedCommonTermsQuery(highFreqOccur, lowFreqOccur, queryBuilder.cutoffFrequency(), queryBuilder.disableCoord(), fieldMapper);
+        expectedQuery.setBoost(queryBuilder.boost());
+        expectedQuery.add(new Term("field", "text"));
+        
+        return expectedQuery;
+    }
+
+    @Override
+    protected void assertLuceneQuery(CommonTermsQueryBuilder queryBuilder, Query query, QueryParseContext context) {
+        if (queryBuilder.queryName() != null) {
+            Query namedQuery = context.copyNamedFilters().get(queryBuilder.queryName());
+            assertThat(namedQuery, equalTo(query));
+        }
+    }
+
+    @Test
+    public void testValidate() {
+    }
```

Comment:
do we need to test validation or not? If so we need some code here, otherwise we can drop the method...